pkg/controller/job/job_controller.go

```diff
@@ -902,9 +902,20 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 
 	var manageJobErr error
 
+	// This is the starting point for the Jobs evaluation steps.
+	// As a Job any kind of criteria evaluation principal, the negative ones (e.g., .spec.backoffLimit, and .spec.activeDeadlineSeconds)
+	// are respected rather than the positive ones (e.g., .spec.completions, and .spec.successPolicy).
+	// And the Job completion (succeeded) can be observed by the existence of the completionTime (.status.completionTime)
+	// instead of succeeded state (.status.succeeded).
+	//
+	// 1. Preserve the existing Job state.
```

Comment:
This seems quite verbose and reduces the chance someone will actually read it. 
Maybe we can have something more light weight, like:

```
This is the starting point for evaluating the end state of the Job. 
Note that in some cases a Job could satisfy multiple checks at the same time, and so we need to order them. 
We implement the following order of checks:
1. pre-existing SuccessCriteriaMet and FailureTarget are read and respected
2. failure scenarios (ActiveDeadlineSeconds exceeded, BackoffLimitExceeded, PodFailurePolicy, etc.)
3. success scenarios (Completions reached or SuccessPolicy met)
```
I also don't have a strong opinion here, but I don't see value for enumerating the checks like "1", "2.1", etc. It adds overhead on maintenance in the future, and it is pretty obvious what the code does. In my view it is more important to capture the overall decision (as above) rather than annotate code line-by-line.

wdyt @atiratree @soltysh ?

Reply:
I agree the intro here could be shorter, something like quick:
```
1. pre-existing SuccessCriteriaMet and FailureTarget are read and respected
2. failure scenarios
3. success scenarios
```
than in the appropriate places you can describe exactly which if stands for which failure case. See my comments below.

Reply:
This sgtm too

Reply:
That makes sense. Let me make this shorter, then add more details to each block.

Reply:
+1 to the above and making sure the comments don't drift over time. Having the prefix in the same form like we do now `Evaluate the negative criteria.` should be enough instead of the numbering. Or just `Negative Criteria: X` ?

Reply:
> Having the prefix in the same form like we do now Evaluate the negative criteria. should be enough instead of the numbering. 

I'm leaning toward numbering since I believe that breaking the ordering implies breaking specifications or that the external developer needs to change the ordering in their controllers.

So, I think the numbering would be better for finding the changed evaluation order in the future. 
@atiratree WDYT?

Reply:
Done.

Reply:
sgtm